internal/controller/postgrescluster/controller.go

```diff
@@ -344,7 +345,7 @@ func (r *Reconciler) Reconcile(
 	if err == nil {
 		exporterQueriesConfig, err = r.reconcileExporterQueriesConfig(ctx, cluster)
 	}
-	if err == nil {
+	if err == nil && !feature.Enabled(ctx, feature.OpenTelemetryMetrics) {
```

Comment:
So I think that you actually want to do what you did for `reconcileExporterQueriesConfig`, which is don't guard here, but inside the method, check for the OpenTelemetryMetrics feature being enabled and if so delete the configmap.